app/addons/documents/queryoptions/queryoptions.react.jsx

```diff
@@ -292,93 +294,126 @@ A key value is the first parameter emitted in a map function. For example emit("
   });
 
   var QueryOptionsController = React.createClass({
-    getStoreState: function () {
-      return {
-        includeDocs: store.includeDocs(),
-        showBetweenKeys: store.showBetweenKeys(),
-        showByKeys: store.showByKeys(),
-        betweenKeys: store.betweenKeys(),
-        byKeys: store.byKeys(),
-        descending: store.descending(),
-        skip: store.skip(),
-        limit: store.limit(),
-        showReduce: store.showReduce(),
-        reduce: store.reduce(),
-        groupLevel: store.groupLevel()
-      };
+
+    getWrap: function () {
+      return connectToStores(TrayWrapper, [store], function () {
+
+        return {
+          includeDocs: store.includeDocs(),
+          showBetweenKeys: store.showBetweenKeys(),
+          showByKeys: store.showByKeys(),
+          betweenKeys: store.betweenKeys(),
+          byKeys: store.byKeys(),
+          descending: store.descending(),
+          skip: store.skip(),
+          limit: store.limit(),
+          showReduce: store.showReduce(),
+          reduce: store.reduce(),
+          groupLevel: store.groupLevel(),
+          contentVisible: store.getTrayVisible(),
+          queryParams: store.getQueryParams()
+        };
+      });
     },
 
-    getInitialState: function () {
-      return this.getStoreState();
+    render: function () {
+      var TrayWrapper = this.getWrap();
+      return (
+        <TrayWrapper>
+          <QueryTray contentVisible={false} />
+        </TrayWrapper>
+      );
+    }
+  });
+
+  var QueryTray = React.createClass({
+
+    propTypes: {
+      contentVisible: React.PropTypes.bool.isRequired
+    },
+
+    runQuery: function (e) {
+      e.preventDefault();
+
+      Actions.runQuery(this.props.queryParams);
+      Actions.toggleQueryBarVisibility(!this.props.contentVisible);
+    },
+
+    toggleTrayVisibility: function () {
+      Actions.toggleQueryBarVisibility(!this.props.contentVisible);
     },
 
     componentDidMount: function () {
-      store.on('change', this.onChange, this);
+      $('body').on('click.QueryTray', function (e) {
+        if ($(e.target).parents('#query-options').length) {
```

Comment:
`.closest()` is a little more performant (just finds the first parent that matches the selector, rather than all.